### PR TITLE
add role assignment shortcut

### PR DESCRIPTION
**Pull request by jordan13game**

### CODE REVIEW

File: lib/apiScenario/apiScenarioLoader.ts [hunk 1]
```diff
@@ -705,6 +705,7 @@ export class ApiScenarioLoader implements Loader<ScenarioDefinition> {
       outputVariables: rawStep.outputVariables ?? {},
       roleAssignment: rawStep.roleAssignment,
       ...convertVariables(rawStep.variables),
+      authentication: ctx.scenario?.authentication ?? ctx.scenarioDef.authentication,
     };
     // TODO
```
Comment (leniatgh) [inline lib/apiScenario/apiScenarioLoader.ts]: this TODO can be deleted

File: lib/apiScenario/apiScenarioRunner.ts [hunk 2]
```diff
@@ -194,23 +197,77 @@ export class ApiScenarioRunner {
...
+      AzureBuiltInRoles.find((r) => r.roleName === step.roleAssignment.roleName)?.roleDefinitionId;
+
+    if (roleId === undefined) {
+      throw new Error(
+        `Cannot find role definition id for role name ${step.roleAssignment.roleName}`
+      );
+    }
+
+    const req: ApiScenarioClientRequest = {
+      host: DEFAULT_ARM_ENDPOINT,
```
Comment (leniatgh) [inline lib/apiScenario/apiScenarioRunner.ts]: host should be an option

File: lib/apiScenario/apiScenarioRunner.ts [hunk 3]
```diff
@@ -194,23 +197,77 @@ export class ApiScenarioRunner {
...
+      method: "PUT",
+      path: "/$(scope)/providers/Microsoft.Authorization/roleAssignments/$(roleAssignmentName)",
+      pathParameters: parameters,
+      headers: {},
+      query: { "api-version": "2022-04-01" },
```
Comment (leniatgh) [inline lib/apiScenario/apiScenarioRunner.ts]: better define a global string constant for the api-version

File: lib/apiScenario/apiScenarioRunner.ts [hunk 4]
```diff
@@ -194,23 +197,77 @@ export class ApiScenarioRunner {
...
+    await this.client.sendRestCallRequest(req, newStep, env);
+  }
+
   private async executeRestCallStep(step: StepRestCall, env: VariableEnv) {
     let req: ApiScenarioClientRequest = {
       host: "",
-      method: step.operation._method.toUpperCase() as HttpMethods,
-      path: step.operation._path._pathTemplate.replace(/{([a-z0-9-_$]+)}/gi, (_, p1) => `$(${p1})`),
+      method: step.operation!._method.toUpperCase() as HttpMethods,
+      path: step.operation!._path._pathTemplate.replace(/{([a-z0-9_$]+)}/gi, (_, p1) => `$(${p1})`),
```
Comment (leniatgh) [inline lib/apiScenario/apiScenarioRunner.ts]: path parameter may contain `-`

File: lib/apiScenario/apiScenarioRunner.ts [hunk 5]
```diff
@@ -194,23 +197,77 @@ export class ApiScenarioRunner {
...
   }
 
+  private async executeArmRoleAssignmentStep(step: StepRoleAssignment, env: VariableEnv) {
+    const parameters = {
+      scope: step.roleAssignment.scope,
+      roleAssignmentName: "{{$guid}}",
+      "api-version": "2022-04-01",
+    };
+
+    const roleId =
```
Comment (leniatgh) [inline lib/apiScenario/apiScenarioRunner.ts]: `roleId` not being used? better rename to `roleDefinitionId`

File: lib/apiScenario/apiScenarioRunner.ts (hunk 2 above)
Comment (jordan13game) [reply lib/apiScenario/apiScenarioRunner.ts]: it's required in ApiScenarioClientRequest

**Review by leniatgh — APPROVED** [review]
looks great, please bump version and update Changelog

**Pull request merged**